### DISCUSSION TITLE
Error when using Azure Blob as work location

### DISCUSSION THREAD

**Discussion by kumarusa2020** [nextflow-io/nextflow#2454] [Q&A] [answered]
Hello members, I am trying a sample nextflow script using Azure Blob storage for work directory. I am getting an error and couldn't understand if my setup has anything to do, appreciate your response. Here is the error when trying to run my sample script connecting to Azure Blob storage. PLEASE HELP!!

`nextflow run tutorial.nf -w az://nextflow-work/work`

### **Script: tutorial.nf**
```
#!/usr/bin/env nextflow

params.str = 'Hello world!'

process splitLetters {

    output:
    file 'chunk_*' into letters

    """
    printf '${params.str}' | split -b 6 - chunk_
    """
}


process convertToUpper {

    input:
    file x from letters.flatten()

    output:
    stdout result

    """
    cat $x | tr '[a-z]' '[A-Z]'
    """
}

result.view { it.trim() }

```
### **Error:** 
![image](https://user-images.githubusercontent.com/72895248/142486683-d2abf7b6-5dda-476d-b134-83290bb9fcf9.png)

**Comment by manuelesimi** [comment, marked as answer]
Hi,

As you can see from the output, you are not submitting your tasks to Azure Batch (`executor > local`). You can't use a blob storage as workdir without using Azure Batch as executor.

manuele
